utils/build-mlir-aie-local.sh

```diff
@@ -8,52 +8,54 @@
 # 
 ##===----------------------------------------------------------------------===##
 #
-# This script build mlir-aie given the <sysroot dir>, <llvm dir> and 
-# <cmakeModules dir>. Assuming they are all in the same subfolder, it would
-# look like:
+# This script build mlir-aie given the <llvm dir> and <libxaie dir>. Assuming 
+# they are all in the same subfolder, it would look like:
 #
-# build-mlir-aie-local.sh <llvm dir> <cmakeModules dir> 
-#     <mlir-aie dir> <build dir> <install dir> <libxaie dir>
+# build-mlir-aie-local.sh <llvm dir> <libxaie dir> <mlir-aie dir> <build dir> 
+#     <install dir>
 #
-# e.g. build-mlir-aie-local.sh /scratch/llvm 
-#     /scratch/cmakeModules/cmakeModulesXilinx
+# e.g. build-mlir-aie-local.sh /scratch/llvm /opt/xaiengine/
 #
+# <libxaie dir>  - libxaie installation dir, default is '/opt/xaiengine'
 # <mlir-aie dir> - optional, mlir-aie repo name, default is 'mlir-aie'
 # <build dir>    - optional, mlir-aie/build dir name, default is 'build'
 # <install dir>  - optional, mlir-aie/install dir name, default is 'install'
-# <libxaie dir>  - optional, libxaie installation dir, default is 
-#     '/opt/xaiengine'
 #
 ##===----------------------------------------------------------------------===##
 
 if [ "$#" -lt 2 ]; then
-    echo "ERROR: Needs at least 2 arguments for <llvm dir> and <cmakeModules dir>."
+    echo "ERROR: Needs at least 2 arguments for <llvm dir> and <libxaie dir>."
     exit 1
 fi
+
 LLVM_DIR=`realpath $1`
-CMAKEMODULES_DIR=`realpath $2`
+LibXAIE_x86_64_DIR=`realpath ${2:-"/opt/xaiengine/"}`
+
+MLIR_AIE_DIR=${3:-"mlir-aie"}
+BUILD_DIR=${4:-"build"}
+INSTALL_DIR=${5:-"install"}
 
-LIBXAIE_DIR=`realpath ${3:-"/opt/xaiengine/"}`
-MLIR_AIE_DIR=${4:-"mlir-aie"}
-BUILD_DIR=${5:-"build"}
-INSTALL_DIR=${6:-"install"}
+CMAKEMODULES_DIR=`realpath $MLIR_AIE_DIR/cmake`
 
 mkdir -p $MLIR_AIE_DIR/$BUILD_DIR
 mkdir -p $MLIR_AIE_DIR/$INSTALL_DIR
 cd $MLIR_AIE_DIR/$BUILD_DIR
+set -o pipefail
+set -e
 
 cmake -GNinja \
     -DCMAKE_C_COMPILER=clang \
     -DCMAKE_CXX_COMPILER=clang++ \
     -DLLVM_DIR=${LLVM_DIR}/build/lib/cmake/llvm \
     -DMLIR_DIR=${LLVM_DIR}/build/lib/cmake/mlir \
-    -DLibXAIE_x86_64_DIR=${LIBXAIE_DIR} \
-    -DCMAKE_MODULE_PATH=${CMAKEMODULES_DIR}/ \
+    -DLibXAIE_x86_64_DIR=${LibXAIE_x86_64_DIR} \
+    -DCMAKE_MODULE_PATH=${CMAKEMODULES_DIR}/modulesXilinx \
     -DCMAKE_INSTALL_PREFIX="../${INSTALL_DIR}" \
-    -DBUILD_SHARED_LIBS=off \
     -DCMAKE_BUILD_TYPE=Release \
     -DLLVM_ENABLE_ASSERTIONS=ON \
     -DAIE_ENABLE_BINDINGS_PYTHON=ON \
+    "-DAIE_RUNTIME_TARGETS=x86_64;aarch64" \
+    -DAIE_RUNTIME_TEST_TARGET=aarch64 \
```

Comment:
This script builds mlir-aie targetting X86, and the cross-build-mlir-aie.sh builds mlir-aie targeting aarch64. There is no file named build-mlir-aie-pcie.sh.

Reply:
sorry, I misread the name. My comment applies to build-mlir-air.sh vs build-mlir-air-pcie.h. It seems like those can be combined.

Why are we hard coding setting the test target to aarch64?

Reply:
This line is copied over from mlir-aie's x86 compilation scrip: https://github.com/Xilinx/mlir-aie/blob/08aff39136b2c797d0f01915f11de4a3ff1edc57/utils/build-mlir-aie.sh#L50. My knowledge of mlir-aie's build flow is limited, and I don't know for sure why they test against aarch64 rather than x86...